examples/multiple_viewer_widget.py

```diff
@@ -0,0 +1,364 @@
+    """move the cross to the mouse position"""
+
+    print(viewer.title, viewer.cursor.position)
+    # FIXME times to time wrong viewer is selected
+
+    viewer.dims.current_step = tuple(
+        np.round(
+            [
+                max(min_, min(p, max_)) / step
+                for p, (min_, max_, step) in zip(
+                    viewer.cursor.position, viewer.dims.range
+                )
+            ]
+        ).astype(int)
+    )
+
+
+action_manager.register_action(
+    name='napari:move_point',
+    command=center_cross_on_mouse,
+    description='Move dims point to mouse position',
+    keymapprovider=ViewerModel,
+)
+
+action_manager.bind_shortcut('napari:move_point', 'C')
+
+
+class own_partial:
+    """
+    Workaround for deepcopy do not copy partial functions
+    (Qt widgets are not serializable)
+    """
+
+    def __init__(self, func, *args, **kwargs):
+        self.func = func
+        self.args = args
+        self.kwargs = kwargs
+
+    def __call__(self, *args, **kwargs):
+        return self.func(*(self.args + args), **{**self.kwargs, **kwargs})
+
+    def __deepcopy__(self, memodict={}):
+        return own_partial(
+            self.func,
+            *deepcopy(self.args, memodict),
+            **deepcopy(self.kwargs, memodict),
+        )
+
+
+class CrossWidget(QCheckBox):
+    """
+    Widget to control the cross layer. because of the performance reason
+    the cross update is throttled
+    """
+
+    def __init__(self, viewer: napari.Viewer):
+        super().__init__("Add cross layer")
+        self.viewer = viewer
+        self.setChecked(False)
+        self.stateChanged.connect(self._update_cross_visibility)
+        self.layer = None  # Vectors(name=".cross", ndim=self.viewer.dims.ndim)
+        self.viewer.dims.events.order.connect(self.update_cross)
+        self.viewer.dims.events.ndim.connect(self._update_ndim)
+        self.viewer.dims.events.current_step.connect(self.update_cross)
+
+    def _update_ndim(self, event):
+        if self.layer in self.viewer.layers:
+            self.viewer.layers.remove(self.layer)
+        self.layer = Vectors(name=".cross", ndim=event.value)
+        self.layer.edge_width = 1.5
+        self.update_cross()
+
+    def _update_cross_visibility(self, state):
+        if state:
+            self.viewer.layers.append(self.layer)
+        else:
+            self.viewer.layers.remove(self.layer)
+        self.update_cross()
+
+    @qthrottled
+    def update_cross(self):
+        if self.layer not in self.viewer.layers:
+            return
+        # self.viewer.layers.remove(self.layer)
+        extent_list = [
+            layer.extent
+            for layer in self.viewer.layers
+            if layer is not self.layer
+        ]
+        extent = Extent(
+            data=None,
+            world=self.viewer.layers._get_extent_world(extent_list),
+            step=self.viewer.layers._get_step_size(extent_list),
+        )
```

Comment:
This is a fragile part as it uses a private API, But for me, it is required as technical layers should not impact the calculation of self size and steep.

Reply:
this calculated area occupied by your data. In your code:
https://github.com/deep-finder/napari-deepfinder/blob/c03e984f12fb7f8a992237329bfd63641d2d774c/src/napari_deepfinder/_orthoview_widget.py#L301

you calculate the data range on the startup of ortho viewer. In this example, this range will be updated on data load/dump/scale update. The general problem with using self.main_view.dims.range is that when you add a layer with a cross it will affect the result of this. 

In this part of the code, the viewer is calculating the same parameters but only for the subset of layers that are real data.